concordium-std/src/traits.rs

```diff
@@ -220,3 +220,43 @@ pub trait ExpectNoneReport {
     /// [fail](macro.fail.html) with the given message, instead of `panic`.
     fn expect_none_report(self, msg: &str);
 }
+
+/// The `SerialCtx` trait provides a means of writing structures into byte-sinks
+/// (`Write`) using contextual information.
+/// The contextual information is:
+///
+///   - `size_length`: The number of bytes used to record the length of the
+///     data.
+pub trait SerialCtx {
+    /// Attempt to write the structure into the provided writer, failing if
+    /// if the length cannot be represented in the provided `size_length` or
+    /// only part of the structure could be written.
+    ///
+    /// NB: We use Result instead of Option for better composability with other
+    /// constructs.
+    fn serial_ctx<W: Write>(
+        &self,
+        size_length: schema::SizeLength,
+        out: &mut W,
+    ) -> Result<(), W::Err>;
+}
+
+/// A wrapper trait for 'Deserial' with additional contextual information,
+/// namely the size_length and ensure_ordered properties.
+/// The `DeserialCtx` trait provides a means of reading structures from
+/// byte-sinks (`Read`) using contextual information.
```

Comment:
I suppose this should be byte-sources, not sinks.

Reply:
Ah okay. I also corrected it in the `Deserial` trait documentation.